### PR TITLE
Improve Windows Compatibility(for lib/THNN)

### PR DESCRIPTION
**Pull request by peterjc123**
Win64 support for lib/THNN

### CODE REVIEW

**Review by fmassa — CHANGES_REQUESTED** [review]
Thanks a lot for the PR!
This looks good for the most part. I have some comments.
I think the best before merging this in would be to first have TH merged in master, and then we run the contribuild again for THNN (but merging TH might break other things, to see).

File: torch/lib/THNN/generic/TemporalRowConvolution.c [hunk 1]
```diff
@@ -73,15 +73,15 @@ static void THNN_(unfolded_acc_row)(
...
+		size_t ix = 0;
 
 		for (kw = 0; kw < kW; kw++) {
 			real *src = finput_data
 			            + c * (kW * nOutputFrame)
 			            + kw * (nOutputFrame);
 			real *dst = input_data + c * (nInputFrame);
 
-			ix = (long long)(kw);
+			ix = (size_t)(kw);
```
Comment (fmassa) [inline torch/lib/THNN/generic/TemporalRowConvolution.c]: I think you missed this casting here

File: torch/lib/THNN/generic/THNN.h [hunk 2]
```diff
@@ -48,7 +48,7 @@ TH_API void THNN_(ClassNLLCriterion_updateOutput)(
           bool sizeAverage,            // if true, the loss will be normalized by batch size and class weights
           THTensor *weights,           // [OPTIONAL] class weights
           THTensor *total_weight,      // [BUFFER]
-          long ignore_index);          // target index to ignore (loss = 0, gradInput = 0)
+          int64_t ignore_index);          // target index to ignore (loss = 0, gradInput = 0)
```
Comment (fmassa) [inline torch/lib/THNN/generic/THNN.h]: nit: the comment is not aligned anymore with the other comments. Not important, but mentioning anyway

File: torch/lib/THNN/generic/THNN.h [hunk 3]
```diff
@@ -57,7 +57,7 @@ TH_API void THNN_(ClassNLLCriterion_updateGradInput)(
...
```
Comment (fmassa) [inline torch/lib/THNN/generic/THNN.h]: same here

File: torch/lib/THNN/generic/THNN.h [hunk 4]
```diff
@@ -67,7 +67,7 @@ TH_API void THNN_(SpatialClassNLLCriterion_updateOutput)(
...
```
Comment (fmassa) [inline torch/lib/THNN/generic/THNN.h]: and here

File: torch/lib/THNN/generic/THNN.h [hunk 5]
```diff
@@ -76,7 +76,7 @@ TH_API void THNN_(SpatialClassNLLCriterion_updateGradInput)(
...
```
Comment (fmassa) [inline torch/lib/THNN/generic/THNN.h]: and here.

**Review by fmassa — APPROVED** [review]
I added some minor comments that don't need to be addressed.
THNN builds without warnings in Unix.
I think this is ready for merge! But let's hold until the other PRs are reviewed so that we can merge them all at once.

Thanks a lot!

**Comment by soumith** [comment]
4 down. thank you @peterjc123 !!!!

**Pull request merged**